### ISSUE TITLE
[UX/Accessibility] Contrasting units/titles

### ISSUE THREAD

**Issue by JorenC** [zond/godip#71]
The darker colours (purple, blue) fail the contrast checks with the black units and text on top of the map. For people with bad eyes, it can be hard to see where the units actually are when the nation colour is closer to black. 

Fix:
1. (code change only) Calculate the contrasting colour (as done previously in DipAct) and then make the border of the units white instead of black. This would be a huge accessibility improvement already.

2. (map & code change): Also allow the SC's to be filled (transparent) black or white depending on occupant colour. Likely quite some effort for relatively small gain.

3. Add the ID for titles (might need 'trace outlines') and let those also be coloured. This will likely be overkill in terms of effort/gain (as users can technically press a region and get the name from the order, or an earlier turn).

**Comment by zond** [comment]
I really think this issue should be in the client, not the adjudicator.

@JorenC, I suggest you open it in the client if you still feel it's important - I'll close it here, OK?

**Comment by tttppp** [comment]
Some of these options need changes in bodily to support. For example the region names will need IDs.

It's not clear to me how to handle text that spans multiple regions. Maybe it should be split into two paths and then each have the ID of the region they are on top of?
Anyway I don't think this is going to happen any time soon.

**Comment by zond** [comment]
I think the title and SCs blobs are the least important of this, since they don't change between phases.

The unit and order shadowing/outline can easily change for dark units, but it's a client issue. Or, maybe we could change dippymap.js so all clients get the update?

**Comment by zond** [comment]
I just realized - can't we shadow the SC blobs and province titles in the map bg color? That way the shadow would only be visible when the provinces were filled with a different color.

Or would that be ugly?

Since this discussion got lively again, maybe it was a mistake to close the issue?

**Comment by JorenC** [comment]
I think title is not so important, but SCs might be. If we start with a dark colour, I never know where the SC's are...

This is an accessibility issue, I don't think it has super high priority. 

I also wouldn't go with light shadows, it's really weird. I think white border around the dark units would work better.

1) SC transparent white instead of transparent black
2) Unit white outline instead of black.

The rest I think we can leave

**Comment by tttppp** [comment]
The issue with the title and SC blobs is that the background colour changes between phases.On most maps the SCs are a fixed grey, but JorenC wanted to change these to be alpha black. I think classical is now like this. This would need dynamically updating to alpha white for dark backgrounds.

**Comment by zond** [comment]
I agree with @JorenC's comment - let's do that :)

**Comment by zond** [comment]
I'll make issues in diplicity and dipact about this, and close it here.
